src/osdc/Objecter.cc

```diff
@@ -3140,8 +3140,10 @@ void Objecter::_finish_op(Op *op, int r)
 
   // op->session->lock is locked unique or op->session is null
 
-  if (!op->ctx_budgeted && op->budgeted)
-    put_op_budget(op);
+  if (!op->ctx_budgeted && op->budget >= 0) {
```

Comment:
no, this is still buggy. again, what if the op has no budget? in that case, the op->budget is -1, and "op_throttle_ops.put(1);" is not called in that case. op_throttle_ops will not reduced after the op w/o budget completes.

Reply:
calc_op_budget return value >=0.  Only budget >=0, it call op_throttle_ops.put(1).